src/sinks/new_relic/model.rs

```diff
@@ -57,39 +41,70 @@ impl TryFrom<Vec<Event>> for MetricsApiModel {
+                        bt.insert(key, Value::from(value));
+                    }
+                    Some(Value::from(bt))
+                } else {
+                    None
+                };
+
+                let mut metric_data = KeyValData::new();
+
+                if let MetricValue::Gauge { value } | MetricValue::Counter { value } = data.value {
+                    metric_data.insert("name".to_owned(), Value::from(series.name.name));
+                    metric_data.insert(
+                        "value".to_owned(),
+                        Value::from(
+                            NotNan::new(value)
+                                .map_err(|_| NewRelicSinkError::new("NaN value not supported"))?,
+                        ),
+                    );
+                    metric_data.insert(
+                        "timestamp".to_owned(),
+                        if let Some(ts) = data.time.timestamp {
+                            Value::from(ts.timestamp())
+                        } else {
+                            Value::from(DateTime::<Utc>::from(SystemTime::now()).timestamp())
+                        },
+                    );
+                    if let Some(attr) = attr {
+                        metric_data.insert("attributes".to_owned(), attr);
                     }
-                    MetricValue::Counter { value } => {
-                        metric_array.push((
-                            Value::from(metric.name().to_owned()),
-                            Value::from(
-                                NotNan::new(*value).map_err(|_| {
-                                    NewRelicSinkError::new("NaN value not supported")
-                                })?,
-                            ),
-                            Value::from(metric.timestamp()),
-                        ));
+                }
+
+                match (data.value, data.kind) {
+                    (MetricValue::Counter { .. }, MetricKind::Incremental) => {
+                        if let Some(interval_ms) = data.time.interval_ms {
+                            metric_data.insert(
+                                "interval.ms".to_owned(),
+                                Value::from(interval_ms.get() as i64),
+                            );
+                        } else {
+                            // Incremental counter without an interval is worthless, skip this metric
+                            continue;
+                        }
+                        metric_data.insert("type".to_owned(), Value::from("count"));
                     }
-                    _ => {
-                        // Unrecognized metric type
+                    (
+                        MetricValue::Gauge { .. } | MetricValue::Counter { .. },
+                        MetricKind::Absolute,
+                    ) => {
+                        metric_data.insert("type".to_owned(), Value::from("gauge"));
                     }
+                    _ => {}
```

Comment:
Should this also `continue` here, or is it valid to add metrics without a `type`? If the former, then there appears to be an opportunity for refactoring the common insertion of the type out of the match.

Reply:
That's incorrect, thanks for caching it! It shouldn't insert the metric into the vector if it doesn't meet the requirements. Fixed it.